### ISSUE TITLE
Upgrade to vlucas/phpdotenv:^3.3

### ISSUE THREAD

**Issue by GrahamCampbell** [silverstripe/silverstripe-framework#8764]
V3 has lots of improvements including controlling if the non-thread-safe put/getenv functions or used, and controlling which superglobals are modified. Here's an example of loading into the server and env superglobals only:

```php
<?php

use Dotenv\Environment\Adapter\EnvConstAdapter;
use Dotenv\Environment\Adapter\ServerConstAdapter;
use Dotenv\Environment\DotenvFactory;
use Dotenv\Dotenv;

$factory = new DotenvFactory([new EnvConstAdapter(), new ServerConstAdapter()]);

Dotenv::create($path, null, $factory)->load();
```

**Comment by robbieaverill** [comment]
cc @silverstripe/core-team (particularly @dhensby and @tractorcow)

**Comment by maxime-rainville** [comment]
Looking back at our commit history, we use to use `vlucas/phpdotenv` but that got remove in favour of `m1/env` in https://github.com/silverstripe/silverstripe-framework/pull/7484

**Comment by GrahamCampbell** [comment]
I don't think it supports multiline strings, nor is it as flexible as phpenv v3.

**Comment by sminnee** [comment]
From memory we switched to m1 due to thread-safety issues with putenv, which it sounds like phpdotenv 3 have solved.

I don't want us to maintain our own env parser, but it sounds like we're relying on m1/env for that. I don't think there's a major motivation to change back, but if people come across bugs in m1 switching back to phpdotenv would be an option.

**Comment by tractorcow** [comment]
`$_ENV` isn't normally reliable in php apps, given many servers don't even automatically-populate it. There were other problems with trying to merge loaded vars into the real environment.

In the end it turned out to be much safer and robust to wrap our overloaded env vars in our `Environment` class. I don't see the need, but I do see additional risks, in trying to migrate back away from that design.
